lib/oper1.g

```diff
@@ -601,6 +601,31 @@ BIND_GLOBAL( "INSTALL_METHOD",
       fi;
     fi;
 
+    if IS_FUNCTION(method) and IsBound(HasNameFunction) and
+      IsBound(TYPE_FUNCTION_WITH_NAME) and IsBound(TYPE_OPERATION_WITH_NAME) and
+      not VAL_GVAR("HasNameFunction")(method) then
```

Comment:
So, I am in principle fine with keeping this as is, but here's an alternative that might perhaps be a bit easier to read, and which would ensure that the handful of methods which are installed before `HasNameFunction` is defined properly get names, too: Right before this function, insert `HasNameFunction := ReturnFalse;`, possibly with a comment explaining what's going on. Then `Unbind(HasNameFunction);` right before `HasNameFunction` is actually defined. And here, we can then simply return to the original form, i.e.:
```
     if IS_FUNCTION(method) and HasNameFunction(method) then
```
and of course similar for the rank function